src/Model/Profile.php

```diff
@@ -902,30 +887,25 @@ public static function getThemeUid(App $a)
 	 */
 	public static function searchProfiles($start = 0, $count = 100, $search = null)
 	{
-		$publish = (DI::config()->get('system', 'publish_all') ? '' : "`publish` = 1");
+		$publish = (DI::config()->get('system', 'publish_all') ? '' : "AND `publish` = 1");
```

Comment:
This won't work without changing all the queries that depend on this variable as well.

Reply:
I have now accepted all your code suggestions, but haven't checked if they cover all cases.